### ISSUE TITLE
Problem with TOC in encrypted pdf file

### ISSUE THREAD

**Issue by jlb6907** [ahrm/sioyek#1216]
Hi,

Sioyek cannot correctly read a table of contents from an encrypted PDF file.

See these links:
- The [python script to create the files](https://github.com/jlb6907/test/blob/main/test_crypt.py)
- The [PDF file without encryption](https://github.com/jlb6907/test/blob/main/sample.pdf)
- The [PDF File with encryption](https://github.com/jlb6907/test/blob/main/sample_encrypt.pdf) (password=mypassword)

TOC of encrypted PDF file can be read correctly with other PDF readers (Acrobat reader, Foxit, ...)

Thanks a lot.

**Comment by jlb6907** [comment]
A little screenshot to see the problem (toc view): 

![ScreenShot](https://github.com/user-attachments/assets/485e49b1-b4fb-4caf-87fa-4e51700b4bbc)

**Comment by ahrm** [comment]
Yes, I can reproduce the issue on the alpha release build but not on the latest commits in the development branch, I guess I have inadvertently fixed the issue.
